kong/pdk/request.lua

```diff
@@ -693,30 +694,69 @@ local function new(self)
   --
   -- If the size of the body is greater than the Nginx buffer size (set by
   -- `client_body_buffer_size`), this function fails and returns an error
-  -- message explaining this limitation.
+  -- message explaining this limitation, unless `max_allowed_file_size`
+  -- is set and larger than the body size buffered to disk.
+  -- Use of `max_allowed_file_size` requires Kong to read data from filesystem
+  -- and has performance implications.
   --
   -- @function kong.request.get_raw_body
   -- @phases rewrite, access, response, admin_api
+  -- @max_allowed_file_size[opt] number the max allowed file size to be read from
   -- @treturn string|nil The plain request body or nil if it does not fit into
   -- the NGINX temporary buffer.
   -- @treturn nil|string An error message.
   -- @usage
   -- -- Given a body with payload "Hello, Earth!":
   --
   -- kong.request.get_raw_body():gsub("Earth", "Mars") -- "Hello, Mars!"
-  function _REQUEST.get_raw_body()
+  function _REQUEST.get_raw_body(max_allowed_file_size)
     check_phase(before_content)
 
     read_body()
 
     local body = get_body_data()
     if not body then
-      if get_body_file() then
+      local body_file = get_body_file()
+      if not body_file then
+        return ""
+      end
+
+      if not max_allowed_file_size then
         return nil, "request body did not fit into client body buffer, consider raising 'client_body_buffer_size'"
+      end
 
-      else
-        return ""
+      local file, err = io.open(body_file, "r")
```

Comment:
IMHO it's safer to read the file content as binary. `local file, err = io.open(body_file, "rb")`. Just to be safe. https://www.lua.org/pil/21.2.2.html